core/jazz_acl/components/casbin.js

```diff
@@ -17,55 +17,185 @@
+    if (isPoliciesRemoved) {
+      // Save the policy back to DB.
+      let savedPolicies = policies.map(async policy => await enforcer.addPolicy(policy.userId, `${serviceId}_${policy.category}`, policy.permission));
+      savedPolicies = await Promise.all(savedPolicies);
+      result.success = await enforcer.savePolicy();
+      
+    } else {
+      result = {
+        "success": false,
+        "error": `Could not remove the policies`
+      };  
+    }
+  } catch(err) {
+    logger.error(err.message);
+    result = {
+      "success": false,
+      "error": err.message
+    };
+  }
+  await conn.close();
+  return result;
 }
 
-async function enforce(policy) {
-
-  //TODO implement the enforce method
-  return true;
+/* Get the policies from casbin given the serviceId*/
+async function getPolicies(serviceId, config) {
+  const values = [`${serviceId}_manage`, `${serviceId}_code`, `${serviceId}_deploy`];
+  const result = await getFilteredPolicy(1, values, config);
+  if (result && result.success) {
+    let filteredPolicies = result.data.filter(el => el.length >=1);
+    if (filteredPolicies.length) {
+      filteredPolicies = filteredPolicies.map(policyArr => policyArr.map(policy => [policy[0], policy[1].split('_')[1], policy[2]]));
+      result.data = filteredPolicies;
+    } else {
+      result.data = [];
+    }
+  }
+  
+  return result;
 }
 
-/* Get the policies from casbin given the value*/
+/* Get the policies from casbin given the values and index*/
 async function getFilteredPolicy(index, values, config) {
-  //TODO Implement getFilteredPolicy
+  let result = {};
+  let conn, enforcer;
+  
+  try {
+    conn = await dbConnection(config);
+    enforcer = await Enforcer.newEnforcer('./config/rbac_model.conf', conn);
+    
+    const promisedPolicies = values.map(async value => await enforcer.getFilteredPolicy(index, value));
+    const policies = await Promise.all(promisedPolicies);
+    result = {
+      "success": true,
+      "data": policies
+    };
+  } catch(err) {
+    logger.error(err.message);
+    result = {
+      "success": false,
+      "error": err.message
+    };
+  }
+  await conn.close();
   return result;
 }
 
-async function removeFilteredPolicy(index, value) {
-  //TODO: implement removeFilteredPolicy
+/* Check permissions for a user */
+async function enforce(userId, serviceId, category, permission, config) {
+  let result = {};
+  
+  try {
+    const conn = await dbConnection(config);
+    const enforcer = await Enforcer.newEnforcer('./config/rbac_model.conf', conn);
+    result.authorized = enforcer.enforce(userId, `${serviceId}_${category}`, permission);
+  } catch(err) {
+    logger.error(err.message);
+    result = {
+      authorized: false,
+      error: err.message
+    }
+  }
 
-  return true;
+  return result;
 }
 
-async function getPermissionForUser(user) {
-  //TODO: implement getPermissionForUser
+/* Remove filtered policy */
+async function removeFilteredPolicy(serviceId, config) {
```

Comment:
Suggestion - We're creating two DB connections during the execution of this function - `removeFilteredPolicy`. Is there a possibility to avoid this & reuse connections?

Reply:
i tried it and i have been getting issues with db connection. it is not closing it properly and even after lambda is executed the connection remains open. I even tried passing opened connection between function which was causing issues as well.